scripts/scr_weapon/scr_weapon.gml

```diff
@@ -1681,6 +1542,167 @@ global.weapons = {
 		"arp": 0,
 		"tags": ["vehicle", "turret", "volkite", "ancient"]
 	},
+// Hireling weapons
+	// Admech
+	"Hellgun": {
+		"abbreviation": "HllGun",
+		"attack": {
+			"standard": 30,
+			"master_crafted": 35,
+			"artifact": 40
+		},
+		"description": "Known as a Hellgun or Hot-shot Lasgun, these high-energy weapons are more potent and destructive than common Lasguns, drawing more power for a more penetrative shot, but also more complex, requiring reinforced barrels, thermal-cooling cells and gyro-stabilized power packs.",
+		"ammo": 20,
+		"range": 12,
+		"spli": 3,
+		"arp": 0,
+		"tags": ["las"],
+		"melee_hands": 0,
+		"ranged_hands": 1,
+	},
+	"Laspistol": {
+		"attack": {
+			"standard": 20,
+			"master_crafted": 30,
+			"artifact": 40
+		},
+		"description": "The Laspistol is the pistol version of the Lasgun and like that weapon fires a coherent beam of energetic photons that can burn through most materials. The Laspistol is powered by a miniature power pack that is usually placed within the grip.",
+		"abbreviation": "Lpstl",
+		"melee_hands": 0,
+		"ranged_hands": 0.25,
+		"ammo": 30,
+		"range": 3.1,
+		"spli": 0,
+		"arp": 0,
+		"tags": ["pistol", "las"],
+	},
+	// Other imperials
+	"Light Bolter": {
+		"abbreviation": "LghtBltr",
+		"attack": {
+			"standard": 35,
+			"master_crafted": 40,
+			"artifact": 45
+		},
+		"description": "A smaller variant of Bolter, intended to be useable by unaugmented humans.",
+		"melee_hands": 0,
+		"ranged_hands": 1,
+		"ammo": 16,
+		"range": 10,
+		"spli": 2,
+		"arp": 0,
+		"tags": ["bolt"],
+		"second_profiles": ["Sarissa"]
+	},
+	"Sarissa": {
+		"abbreviation": "Saris",
+		"attack": {
+			"standard": 25,
+			"master_crafted": 30,
+			"artifact": 35
+		},
+		"description": "A vicious combat attachment that is attached to Bolters, in order to allow them to be used in melee combat.",
+		"melee_hands": 0,
+		"ranged_hands": 0,
+		"ammo": 0,
+		"range": 1,
+		"spli": 1,
+		"arp": 0,
+		"tags": ["axe"],
+	},
+	"Hellrifle": {
+		"description": "Normally used by Radical Inquisitors, it appears an antiquated rifle but fires razor-sharp shards of Daemonic matter.",
+		"abbreviation": "HllRifle",
+		"attack": {
+			"standard": 150,
+			"master_crafted": 160,
+			"artifact": 170
+		},
+		"ammo": 10,
+		"range": 6.1,
+		"spli": 3,
+		"arp": 0,
+		"tags": ["rifle", "arcane"],
+		"ranged_hands": 1,
+		// ... (other attributes)
+	},
```

Comment:
Updated description for this rifle, buffed damage and range, but reduced spli.
```suggestion
	"Hellrifle": {
		"description": "Extremely effective and intricate weapons frowned upon by more Puritanical Inquisitors. Fire shards of an unknown material. Speculated to be daemonic
in origin.",
		"abbreviation": "HllRifle",
		"attack": {
			"standard": 200,
			"master_crafted": 250,
			"artifact": 300
		},
		"ammo": 10,
		"range": 20,
		"spli": 0,
		"arp": 0,
		"tags": ["rifle", "arcane"],
		"ranged_hands": 1,
	},
```

Reply:
I think it needs to have spli at least 1.
I remember back then, having spli as 0 meant weapons would not fire.

Reply:
But, let's see anyway, maybe it has been fixed since then...

Reply:
> I think it needs to have spli at least 1. I remember back then, having spli as 0 meant weapons would not fire.

If so, then Stalker Bolter also has 0 spli. The code seems to either ignore values below 1 or reset them to 1.
Either way, you can fix this on all other weapons in this file, as I can only modify stuff that you've modified before.